src/forms/Button.jsx

```diff
@@ -69,25 +71,38 @@ class Button extends React.PureComponent {
 			</Flex>
 		);
 
+		const ButtonTag = buttonTag;
+		const hrefAttr = buttonTag === 'Link' && buttonTag !== 'button' ? 'to' : 'href';
+		const buttonProps = {
+			className: classNames.button,
+			onClick: onClick,
+			role: buttonTag ? 'button' : null,
+			[hrefAttr]: linkTo,
+			...other
+		};
+
 		return (
-			<button
-				className={classNames.button}
-				onClick={onClick}
-				role='button'
-				{...other}
+			<ButtonTag
+				{...buttonProps}
 			>
 				{ icon ? iconChildren : children }
-			</button>
+			</ButtonTag>
 		);
 	}
 }
 
+Button.defaultProps = {
+	buttonTag: 'button'
+};
+
 Button.propTypes = {
 	reset: PropTypes.bool,
 	fullWidth: PropTypes.bool,
 	primary: PropTypes.bool,
 	small: PropTypes.bool,
 	icon: PropTypes.any,
 	right: PropTypes.bool,
+	buttonTag: PropTypes.oneOf(['button', 'a', 'Link']),
```

Comment:
`Link` is not like the other two options here, `button`, and `a`. The button and anchor strings are lowercase and will be treated by React as the DOM elements they are.

`Link` refers to an actual component, which means you'd either have to:

1. always import `Link` from `react-router` and keep the version of `react-router` in this repo in sync with consumer apps
2. split out `Link` as a separate prop, that accepts `PropTypes.element`.

I'm recommending option (2) here because it removes a dependency on `react-router`, and it would also enable you to remove the `linkTo` prop, as the `Link` component would be passed in whole as a prop:

```
<Button linkComponent={<Link linkTo="lol">My cool link</Link>} />
```

In the end, you would have a prop to accommodate a link component, and a prop that allows users to specify an html element to render. In the event a user passes both, we could favor rendering `Link`.

```
Button.propTypes = {
	...
	htmlElement: PropTypes.oneOf(['button', 'a']),
	linkComponent: PropTypes.element,
	...
};
```

Reply:
You're 100% right, and I'm going with option 2